<a href="https://colab.research.google.com/github/sittirahmahsindar/Proyek-Data-Wrangling/blob/main/Teknik_pengambilan_sampel_dan_data_wrangling_ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

# membaca file csv
df = pd.read_csv('/content/diabetes.csv')

In [ ]:
df.head()  # menampilkan 5 data pertama


,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,0,33.6,0.627,50,1
1,1,85,66,29,0,26.6,0.351,31,0
2,8,183,64,0,0,23.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1


In [ ]:
df.info() # informasi tipe data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [ ]:
df.shape # jumlah baris dan kolom

(768, 9)

Dataset **Diabetes (Pima Indians Diabetes Database)** terdiri dari **768 baris** dan **9 kolom**. Setiap baris merepresentasikan satu individu atau pasien yang menjadi objek penelitian. Dari total 9 kolom tersebut, delapan kolom pertama merupakan variabel independen (fitur), yaitu Pregnancies, Glucose, BloodPressure, SkinThickness, Insulin, BMI, DiabetesPedigreeFunction, dan Age, sedangkan satu kolom terakhir yaitu Outcome merupakan variabel dependen (target) yang menunjukkan hasil diagnosis diabetes, dengan nilai 0 untuk tidak diabetes dan 1 untuk diabetes.

Berdasarkan **informasi tipe data**, seluruh variabel dalam dataset ini bertipe numerik, yang terdiri dari 7 kolom bertipe **int64** dan 2 kolom bertipe **float64**. Kolom yang bertipe int64 meliputi Pregnancies, Glucose, BloodPressure, SkinThickness, Insulin, Age, dan Outcome, sedangkan kolom BMI dan DiabetesPedigreeFunction bertipe float64 karena mengandung nilai desimal. Selain itu, seluruh kolom memiliki 768 data non-null, yang berarti tidak terdapat nilai kosong (missing values) dalam dataset ini. Struktur data sepenuhnya yaitu numerik.

In [17]:
import json

data = [
    {
        "Nama": "Alya",
        "Umur": 20,
        "Jurusan": "Ilmu Komputer",
        "IPK": 3.75
    },
    {
        "Nama": "Adi",
        "Umur": 21,
        "Jurusan": "Informatika",
        "IPK": 3.60
    },
    {
        "Nama": "Rahmah",
        "Umur": 19,
        "Jurusan": "Sains Data",
        "IPK": 3.85
    }
]

print(json.dumps(data, indent=4))

[
    {
        "Nama": "Alya",
        "Umur": 20,
        "Jurusan": "Ilmu Komputer",
        "IPK": 3.75
    },
    {
        "Nama": "Adi",
        "Umur": 21,
        "Jurusan": "Informatika",
        "IPK": 3.6
    },
    {
        "Nama": "Rahmah",
        "Umur": 19,
        "Jurusan": "Sains Data",
        "IPK": 3.85
    }
]


In [18]:
df.to_csv("data_mahasiswa", index=False)
df

,Transaction_ID,Customer_Name,Product_Category,Quantity,Price,Total_Amount,Transaction_Date,Payment_Method
0,TRX001,Andi Saputra,Electronics,2,3500000,7000000,2025-01-10,Credit Card
1,TRX002,Siti Rahma,Fashion,3,250000,750000,2025-01-12,E-Wallet
2,TRX003,Budi Santoso,Home & Living,1,1200000,1200000,2025-01-15,Bank Transfer
